# HG changeset patch
# Parent  37abecb2b0d3417349a2081a541cfbe4609f6892
Improve error handling in lgtd_tests_make_temp_dir

Hopefully let me solve those segfaults in the unit-tests of the pipe
module.

diff --git a/tests/core/tests_utils.c b/tests/core/tests_utils.c
--- a/tests/core/tests_utils.c
+++ b/tests/core/tests_utils.c
@@ -180,10 +180,13 @@
 {
     char buf[PATH_MAX] = { 0 };
     int n = snprintf(buf, sizeof(buf), "%s/lightsd.test.XXXXXXXX", P_tmpdir);
-    if (n >= (int)sizeof(buf)) {
-        errx(1, "cannot allocate temporary directory");
+    if (n >= (int)sizeof(buf) || n < 0) {
+        lgtd_errx(1, "cannot allocate temporary directory");
     }
-    return strdup(mkdtemp(buf));
+    if (!mkdtemp(buf)) {
+        lgtd_err(1, "cannot create temporary directory");
+    }
+    return strdup(buf);
 }
 
 void
